pkg/termcorder/termcorder.go

```diff
@@ -0,0 +1,210 @@
+package termcorder
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+
+	"github.com/creack/pty"
+	"golang.org/x/term"
+)
+
+//Termcording represent a command/terminal recording
+type Termcording struct {
+	Config *Config
+	cmd    *exec.Cmd
+	out    io.Writer
+}
+
+//Config represents the command configuration
+type Config struct {
+	Filename      string
+	CmdName       string
+	CmdArgs       []string
+	QuietMode     bool
+	Append        bool
+	Interactive   bool
+	LogKeystrokes bool
+	PrintHelp     bool
+}
+
+const defaultFileName = "termcording"
+
+var cli *flag.FlagSet
+
+//Cmd sets the termcording's cmd.
+func Cmd(c *exec.Cmd) func(*Termcording) error {
+	return func(tc *Termcording) error {
+		tc.cmd = c
+		return nil
+	}
+}
+
+//Output sets the termcording's output.
+func Output(w io.Writer) func(*Termcording) error {
+	return func(tc *Termcording) error {
+		tc.out = w
+		return nil
+	}
+}
+
+//NewTermcording returns a pointer to a new variable of type Termcording given a config
+//variable and (functional) options.
+func NewTermcording(c *Config, options ...func(*Termcording) error) (*Termcording, error) {
```

Comment:
Functional options are awesome, but I don't think we should mix them with config structs... they're an alternative way to solve the same problem that config structs solve, aren't they? (In my opinion, a better one.)

Reply:
They indeed are used to solve the same problem. My rationale from using both approaches is from something you said in a recent Code Club video. Something about configurations meaning things that are required vs options being things that are...optional. So I felt that it made sense to seperate them like that. It also comes down to where "Configs" are stored in the struct vs where "options" are. Configs are passed directly to the new `Termcording` value and are things that are set/derived from the command line flags whereas functional options set the other fields of Termcording (for Mocking, for now).